client/sites/components/panel-sidebar/index.tsx

```diff
@@ -1,50 +1,48 @@
 import { Button } from '@wordpress/components';
 import clsx from 'clsx';
-import { navigate } from 'calypso/lib/navigate';
-
+import React from 'react';
+import type { ReactNode } from 'react';
 import './style.scss';
 
-interface PanelSidebarItem {
-	key: string;
-	label: string;
+interface SidebarItemProps {
+	href: string;
+	itemKey: string;
+	selectedItemKey?: string;
+	children: ReactNode;
 }
 
-interface PanelSidebarProps {
-	items: PanelSidebarItem[];
-	selectedItemKey: string;
+export function SidebarItem( { href, itemKey, selectedItemKey, children }: SidebarItemProps ) {
+	return (
+		<li>
+			<Button
+				href={ href }
```

Comment:
This `href` approach is interesting though; maybe we can refactor the existing approach to using this, by passing a base path to `makeSidebar()`. This way we can eliminate the convoluted `switchItem` onClick callback.

Reply:
Refactored this away completely matching pathname to href.